python_modules/libraries/dagster-dbt/dagster_dbt/core/resources_v2.py

```diff
@@ -885,80 +893,53 @@ def _fetch_and_attach_row_count_metadata(
         """Threaded task which fetches row counts for materialized dbt models in a dbt run
         once they are built, and attaches the row count as metadata to the event.
         """
-        adapter = check.not_none(self._dbt_cli_invocation.adapter)
-
-        dbt_resource_props = self._get_dbt_resource_props_from_event(event)
-        is_view = dbt_resource_props["config"]["materialized"] == "view"
-
-        # Avoid counting rows for views, since they may include complex SQL queries
-        # that are costly to execute. We can revisit this in the future if there is
-        # a demand for it.
-        if is_view:
+        if not isinstance(event, (AssetMaterialization, Output)):
             return event
 
-        # If the adapter is DuckDB, we need to wait for the dbt CLI process to complete
-        # so that the DuckDB lock is released. This is because DuckDB does not allow for
-        # opening multiple connections to the same database when a write connection, such
-        # as the one dbt uses, is open.
-        try:
-            from dbt.adapters.duckdb import DuckDBAdapter
+        with pushd(str(self._dbt_cli_invocation.project_dir)):
```

Comment:
I'm afraid that this folder management scheme might bite us in the future... 

In a separate PR, could we consider lumping this `pushd` behavior explicitly when the `.adapter` property of `DbtCliInvocation` is accessed?